components/table/table-wrapper.js

```diff
@@ -156,7 +156,7 @@ export const tableStyles = css`
 	d2l-table-wrapper[sticky-headers]:not([sticky-headers-scroll-wrapper]) .d2l-table > * > tr.d2l-table-header > [sticky],
 	d2l-table-wrapper[sticky-headers]:not([sticky-headers-scroll-wrapper]) .d2l-table > * > tr[header] > .d2l-table-sticky-cell,
 	d2l-table-wrapper[sticky-headers]:not([sticky-headers-scroll-wrapper]) .d2l-table > * > tr[header] > [sticky] {
-		z-index: 3;
+		z-index: 4;
```

Comment:
Hey Noel, do you happen to have a screenshot of the issue you are trying to address? Is your plan to have the app explicitly set a z-index on the dropdown that it's rendering?

Reply:
Yeah whatever info you have about this on-stack use case would be helpful. I'm going to poke around in here and add a visual-diff to exercise the specific case, hopefully avoiding regressions in the future.

Reply:
In some limited contexts, we use the new tables in Grades. We added `sticky-headers` to `d2l-table-wrapper` to see if the sticky headers would work there. It introduced a problem when there were sub-heading cells to a category:

![image](https://github.com/BrightspaceUI/core/assets/5035319/61e1c447-710f-4692-898f-2a915f7a90fd)

The drop-down menu from the category cell is rendered beneath the other header cells.

Reply:
So we added a new style targeting the drop-down's header specifically, lifting it to z-index 3.

![image](https://github.com/BrightspaceUI/core/assets/5035319/10898b4c-071e-44a8-98c7-8e29fc6d5988)

This resolved the overlapping problem.

Reply:
However, raising this cell to z-index 3 put it up above the z-index cells from this core table that also had a z-index of 3.

![image](https://github.com/BrightspaceUI/core/assets/5035319/e91b3993-4fa4-4b7f-bc99-4b1e29b7f30a)

We don't believe the drop-down promoted cells should be promoted above the cells that this table wants to be stuck over top.

Reply:
It would be great if we could find a way to handle this case without the consumer having to override `z-index`. We run into lots of stacking context issues in core components, and having consumers override them is likely to create some pain down the road.

Reply:
Hi! Noel did contact me about this, and a related PR last week:
https://github.com/Brightspace/lms/pull/35974

From what I saw I think we do need the extra z-index space. We might be able to add an extra class in our core styles, something like `.has-dropdown`, and account for how we want that to display (when it's a normal cell, when it's sticky, when it's a header, when it's a header + sticky column...)

This is all a workaround to the more desired improvement of popping dropdowns out of the stacking context, so they're not tied to their parent's z-index.

Reply:
Sounds a little like our list/card shenanigans where we adjust z-index based on contents and events. 😬

Reply:
A bit of other context:

- Grid is overriding some styles [here](https://search.d2l.dev/xref/lms/lp/framework/web/D2L.Web/ui/grid/Css/Grid.css?r=22961f48#20), they all seem unnecessary (duplicates of what's in core already) except lines 47-49
- Noel sent me a link to this other related PR, won't have a chance to look at it today unfortunately: https://github.com/Brightspace/lms/pull/36329

Reply:
@eKoopmans I think you're right, and I plan to update the PR with that change.